esp/GBPlay/main/hardware/wifi.c

```diff
@@ -218,38 +174,73 @@ bool wifi_connect(const char* ssid, const char* password)
         return false;
     }
 
-    ESP_ERROR_CHECK(esp_wifi_set_config(WIFI_IF_STA, &cfg));
-
-    if (esp_wifi_connect() != ESP_OK)
     {
-        return false;
-    }
+        assert(xSemaphoreTakeRecursive(s_wifi_lock, portMAX_DELAY) == pdTRUE);
+
+        bool did_connect = false;
+
+        if (!force && wifi_is_connected())
```

Comment:
Consider the case where the wifi manager finds a network and tries to auto connect to it while the user is trying to connect to a specific network. User-selected networks should be given priority. When the manager task gets the lock, we don't want it to disconnect from what was just chosen.

With `force = false`, we just bail out if there was already a connection. The manager will then see it has nothing left to do and suspend itself again.